packages/web-client/tests/integration/components/card-pay/create-merchant-workflow/merchant-customization-test.ts

```diff
@@ -0,0 +1,273 @@
+      assert.dom(`${PREVIEW} ${MERCHANT}`).containsText('Enter merchant name');
+      assert
+        .dom(`${PREVIEW} ${MERCHANT}`)
+        .hasAttribute('data-test-merchant', 'Enter merchant name');
+      assert.dom(MERCHANT_NAME_FIELD).exists();
+      assert.dom(MERCHANT_ID_FIELD).exists();
+      assert.dom(COLOR_FIELD).exists();
+      assert.dom(SAVE_DETAILS_BUTTON).exists().isDisabled();
+      assert.dom(MANAGER).containsText(layer2AccountAddress);
+    });
+
+    test('It validates the merchant name field and updates the preview', async function (assert) {
+      // enter valid merchant id so we can check that the merchant name being valid will make
+      // the save details button enabled
+      let merchantIdInput = `${MERCHANT_ID_FIELD} input`;
+      await fillIn(merchantIdInput, 'a-valid-id');
+      await waitUntil(
+        () =>
+          (document.querySelector(
+            '[data-test-validation-state-input]'
+          ) as HTMLElement).dataset.testValidationStateInput === 'valid'
+      );
+      assert.dom(SAVE_DETAILS_BUTTON).isDisabled();
+
+      let merchantNameInput = `${MERCHANT_NAME_FIELD} input`;
+
+      // valid
+      await fillIn(merchantNameInput, 'HELLO!');
+      assert.dom(merchantNameInput).hasValue('HELLO!');
+      assert
+        .dom('[data-test-merchant]')
+        .hasAttribute('data-test-merchant', 'HELLO!')
+        .containsText('HELLO!');
+      assert.dom(SAVE_DETAILS_BUTTON).isEnabled();
+
+      for (let invalidEntry of MERCHANT_NAME_INVALID_INPUTS) {
+        await fillIn(merchantNameInput, invalidEntry.value);
+        assert
+          .dom(`${MERCHANT_NAME_FIELD} [data-test-boxel-input-error-message]`)
+          .containsText(invalidEntry.errorMessage);
+        assert.dom(SAVE_DETAILS_BUTTON).isDisabled();
+      }
+    });
+
+    test('It validates the merchant ID field', async function (assert) {
+      let merchantNameInput = `${MERCHANT_NAME_FIELD} input`;
+      await fillIn(merchantNameInput, 'HELLO!');
+      assert.dom(merchantNameInput).hasValue('HELLO!');
+      assert
+        .dom('[data-test-merchant]')
+        .hasAttribute('data-test-merchant', 'HELLO!')
+        .containsText('HELLO!');
+      assert.dom(SAVE_DETAILS_BUTTON).isDisabled();
+
+      let merchantIdInput = `${MERCHANT_ID_FIELD} input`;
+      await fillIn(merchantIdInput, 'a-valid-id');
+      await waitUntil(
+        () =>
+          (document.querySelector(
+            '[data-test-validation-state-input]'
+          ) as HTMLElement).dataset.testValidationStateInput === 'valid'
+      );
+      assert.dom(SAVE_DETAILS_BUTTON).isEnabled();
+
+      for (let invalidEntry of MERCHANT_ID_INVALID_INPUTS) {
+        await fillIn(merchantIdInput, invalidEntry.value);
+        await waitUntil(
+          () =>
+            (document.querySelector(
+              '[data-test-validation-state-input]'
+            ) as HTMLElement).dataset.testValidationStateInput === 'invalid'
+        );
```

Comment:
I was able to remove this `waitUntil` and have the tests still pass because `fillIn` includes a `settled` call? That sometimes doesn’t work if asynchronous processes are happening outside Ember’s test waiters, but did you find that to be the case? (If it is indeed superfluous, that’s likely the case for other instances.)

Reply:
I think I had to add the `waitUntil` because some assertions were failing. I think this might have to do with what you mentioned earlier, resetting to a valid state prior to checking that an invalid one is detected and correctly handled.

Reply:
This one is unnecessary! The rest are necessary as the save details button is disabled while the validation is running.